### ISSUE TITLE
Backups fail when temporary tables disappear before rsync can grab them

### ISSUE THREAD

**Issue by devyn** [EnterpriseDB/barman#293]
I'm using barman in rsync-concurrent mode. Some of my processing systems use temporary tables extensively with very short intervals, and so it happens quite often that barman gets stuck and has to be unblocked manually because a temporary table file didn't exist by the time rsync wanted it:

```
rsync: link_stat "/ephemeral/pgtemp/PG_12_201909212/16402/t3_850798582" failed: No such file or directory (2)
rsync: link_stat "/ephemeral/pgtemp/PG_12_201909212/16402/t3_850798624" failed: No such file or directory (2)
rsync: link_stat "/ephemeral/pgtemp/PG_12_201909212/16402/t3_850798609" failed: No such file or directory (2)
rsync: link_stat "/ephemeral/pgtemp/PG_12_201909212/16402/t3_850798615" failed: No such file or directory (2)
rsync: link_stat "/ephemeral/pgtemp/PG_12_201909212/16402/t3_850798621" failed: No such file or directory (2)
rsync: link_stat "/ephemeral/pgtemp/PG_12_201909212/16402/t3_850798629" failed: No such file or directory (2)
```

Would it make sense to add the `t*_**` pattern for temporary table files to the `EXCLUDE_LIST` in `postgres_plumbing`?

**Comment by barthisrael** [comment]
Hello @devyn,

Although we were unable to reproduce the issue you described, where Barman appears to get stuck due to sessions creating several short-lived temporary files, we would like to clarify and correct our previous statement regarding temporary files.

In our earlier response, we mistakenly conflated two distinct types of temporary data in PostgreSQL: temporary relation files and temporary files created by disk-spilling operations.

To clarify:

* Files under `pgsql_tmp` are used for temporary disk-spill operations.
* Files matching the pattern `t*_**` are temporary relation files.

We are re-opening this issue to further evaluate your suggestion of excluding `t*_**` files, and to assess the feasibility of adding them to the exclusion list. That's actually what `pg_basebackup` does, as we can see here https://github.com/postgres/postgres/blob/REL_17_STABLE/src/backend/backup/basebackup.c#L1337-L1345.

In any case, the current Barman implementation should already be capable of handling these temporary files by ignoring such errors from `rsync`, although it may generate some debug-level log noise in the process.

Best regards,
Israel.

**Comment by barthisrael** [comment]
This issue was automatically closed when a PR was merged into our internal development repository.

This fix might come as part of the next release.
